src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/CreateSdkSymbolsLayout.cs

```diff
@@ -0,0 +1,165 @@
+                        if (peReader.ReadDebugDirectory().Any(entry => entry.Type == DebugDirectoryEntryType.EmbeddedPortablePdb))
+                        {
+                            continue;
+                        }
+
+                        var debugDirectory = peReader.ReadDebugDirectory().First(entry => entry.Type == DebugDirectoryEntryType.CodeView);
+                        var codeViewData = peReader.ReadCodeViewDebugDirectoryData(debugDirectory);
+                        guid = $"{codeViewData.Guid.ToString("N").Replace("-", string.Empty)}";
+                    }
+                    catch (Exception)
+                    {
+                        // Ignore binaries without debug info
+                        continue;
+                    }
+
+                    if (guid != string.Empty)
+                    {
+                        if (!AllPdbGuids.ContainsKey(guid))
+                        {
+                            filesWithoutPDBs.Add(file.Substring(sdkRoot.Length + 1));
+                        }
+                        else
+                        {
+                            // Copy matching pdb to symbols path, preserving sdk binary's hierarchy
+                            string sourcePath = (string)AllPdbGuids[guid]!;
+                            string destinationPath =
+                                file.Replace(sdkRoot, destinationRoot)
+                                    .Replace(Path.GetFileName(file), Path.GetFileName(sourcePath));
+
+                            Directory.CreateDirectory(Path.GetDirectoryName(destinationPath)!);
+                            File.Copy(sourcePath, destinationPath, true);
+                        }
+                    }
+                }
+            }
+
+            return filesWithoutPDBs;
+        }
+
+        public void IndexAllSymbols(string path)
+        {
+            AllPdbGuids = new Hashtable();
+
+            foreach (string file in Directory.GetFiles(path, "*.pdb", SearchOption.AllDirectories))
+            {
+                try
+                {
+                    using var pdbFileStream = File.OpenRead(file);
+                    var metadataProvider = MetadataReaderProvider.FromPortablePdbStream(pdbFileStream);
+                    var metadataReader = metadataProvider.GetMetadataReader();
+                    if (metadataReader.DebugMetadataHeader == null)
+                    {
+                        continue;
+                    }
+
+                    var id = new BlobContentId(metadataReader.DebugMetadataHeader.Id);
+                    string guid = $"{id.Guid:N}";
+                    if (!string.IsNullOrEmpty(guid))
+                    {
+                        if (!AllPdbGuids.ContainsKey(guid))
+                        {
+                            AllPdbGuids.Add(guid, file);
+                        }
+                    }
+                }
+                catch(Exception)
```

Comment:
Are there specific scenarios you expect here or is this a general purpose safegaurd?

Reply:
Just general purpose. We would catch any symbol for which we could not obtain the guid. Originally I was logging this, but there is no value in that.

Reply:
My question was more along the lines if there was a specific exception type(s) we should be catching instead of all. Catching all can potentially mask problems.  This comment applies to the previous catch statement as well.

Reply:
I have considered handling just the specific exceptions... Perhaps we can improve this later.

Reply:
It won't get fixed later 😄 so do it now or never.

Reply:
OK - fixing now 😄

Reply:
Fixed in https://github.com/dotnet/installer/pull/17454/commits/afd434a09fe8b8a68a598741a98df9304259865a